### PR TITLE
Implemented rate-limiting

### PR DESCRIPTION
**Pull request by ericrkuo**

### CODE REVIEW

File: routes/rateLimiter.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const rateLimit = require("express-rate-limit");
+
+const webcamLimiter = rateLimit({
+    windowMs: 60 * 60 * 1000, // 30 min window
+    max: 10, // start blocking after 5 requests
+    message: {errorMsg: "Too many requests from this IP, please try again after an hour"}
```
Comment (ld357) [inline routes/rateLimiter.js:6]: Maybe change the error message to something more friendly?
ex. "Too many requests, please try again after an hour"

File: routes/rateLimiter.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+});
+
+const trackLimiter = rateLimit({
+    windowMs: 10 * 60 * 1000, // 10 min window
+    max: 5, // start blocking after 5 requests
+    message: {errorMsg: "Too many playlists created from this IP, please try again after 10 minutes"}
```
Comment (ld357) [inline routes/rateLimiter.js:12]: Maybe to "Too many playlists created, please try again after 10 minutes."

File: routes/trackRouter.js [hunk 3]
```diff
@@ -1092,7 +1093,7 @@ router.get('/', function (req, res, next) {
 });
 
 // REQUIRES: req.body to contain a list of track URI's in format ["spotify:track:1ue7zm5TVVvmoQV8lK6K2H", ...]
-router.post('/', function (req, res, next) {
+router.post('/', trackLimiter, function (req, res, next) {
```
Comment (ld357) [inline routes/trackRouter.js:1096]: Cool! Do we just have to attach it to the request and it'll automatically apply the limiter?

**Review by ld357 — APPROVED** [review]
AWESOME JOB! Thanks for doing this :) where did you find the library?

File: routes/rateLimiter.js (hunk 1 above)
Comment (ericrkuo) [reply routes/rateLimiter.js:6]: sure

File: routes/trackRouter.js (hunk 3 above)
Comment (ericrkuo) [reply routes/trackRouter.js:1096]: Yup, and it attaches this to each request that we apply the rate limiter. 
image

**Comment by ericrkuo** [comment]
> AWESOME JOB! Thanks for doing this :) where did you find the library?

Oh I searched up rate-limiting for Express.js and this one was commonly used. https://github.com/nfriedly/express-rate-limit

**Pull request merged**